src/mlpack/methods/matrix_completion/matrix_completion.hpp

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file matrix_completion.hpp
+ * @author Stephen Tu
+ *
+ * A thin wrapper around nuclear norm minimization to solve
+ * low rank matrix completion problems.
+ */
+#ifndef __MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_HPP
+#define __MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_HPP
+
+#include <mlpack/core/optimizers/lrsdp/lrsdp.hpp>
+
+namespace mlpack {
+namespace matrix_completion {
+
+/**
+ * This class implements the popular nuclear norm minimization heuristic for
+ * matrix completion problems. That is, given known values M_ij's, the
+ * following optimization problem (semi-definite program) is solved to fill in
+ * the remaining unknown values of X
+ *
+ *   min ||X||_* subj to X_ij = M_ij
+ *
+ * where ||X||_* denotes the nuclear norm (sum of singular values of X).
+ *
+ * For a theoretical treatment of the conditions necessary for exact recovery,
+ * see the following paper:
+ *
+ *   A Simpler Appoarch to Matrix Completion.
+ *   Benjamin Recht. JMLR 11.
+ *   http://arxiv.org/pdf/0910.0651v2.pdf
+ *
+ * An example of how to use this class is shown below:
+ *
+ * @code
+ * size_t m, n;        // size of unknown matrix
+ * arma::umat indices; // contains the known indices [2 x n_entries]
+ * arma::vec values;   // contains the known values [n_entries]
+ *
+ * MatrixCompletion mc(m, n, indices, values);
+ * mc.Recover();
+ * mc.Recovered();     // access completed matrix
+ * @endcode
+ *
+ * @see LRSDP
+ */
+class MatrixCompletion
```

Comment:
You mentioned you're writing an interior-point solver for SDPs... I think once you do that, we can templatize this, which would be really cool:
```
template<typename SDPSolverType = LRSDP> // Or default to the interior point solver, whichever seems to be more robust.
class MatrixCompletion;
```